app-modules/inventory-management/src/Filament/Resources/AssetResource/Pages/ListAssets.php

```diff
@@ -65,8 +68,27 @@ public function table(Table $table): Table
                 TextColumn::make('status.name'),
                 TextColumn::make('location.name'),
                 TextColumn::make('purchase_date')
-                    ->searchable()
-                    ->sortable(),
+                    ->label('Device Age')
+                    ->sortable()
+                    ->formatStateUsing(function (string $state) {
+                        $date = Carbon::parse($state);
+
+                        if ($date->isFuture()) {
+                            return '0 Years 0 Months';
+                        }
+
+                        /** @var User $user */
+                        $user = auth()->user();
+
+                        $diff = $date
+                            ->roundMonth()
+                            ->setTimezone($user->timezone)
+                            ->diff();
+
+                        return $diff->y . ' ' . ($diff->y === 1 ? 'Year' : 'Years') . ' ' .
+                            $diff->m . ' ' . ($diff->m === 1 ? 'Month' : 'Months');
```

Comment:
Can we extract this functionality?

Reply:
Sure, to what? A field? A macro?

Reply:
This is probably one of the few cases that I'd say an attribute actually might make the most sense, unless we foresee this potentially being applicable to other models/fields in the future.

Reply:
Hmm, would an attribute work? The actual field is still `purchase_date` because we need to sort on that so if I put that into an attribute would that mess with the sorting?

Reply:
Actually I see you can pass the column to sort by so it might work.

Reply:
We might not need to do that though, I think you can make the accessor something like `formatted_purchase_date`, access the record inside of the closure in order to access this property. Sort would still be on db column.